### PR TITLE
Feature/pil review

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: pages/common/content/success-messages.js [hunk 1]
```diff
@@ -393,6 +393,45 @@ module.exports = {
...
+      },
+      // TODO: content
+      'recalled-by-applicant': {
+        title: 'Review request recalled',
+        body: 'The review request will need to be resubmitted to be reviewed by an NTCO at your establishment.'
+      },
+      // TODO: content
+      'discarded-by-applicant': {
+        title: 'Review request discarded',
+        body: 'You will need to create a new transfer request if you want to submit these changes again.'
```
Comment (lennym) [inline pages/common/content/success-messages.js]: Transfer request?

File: pages/pil/read/index.js [hunk 2]
```diff
@@ -34,7 +50,7 @@ module.exports = settings => {
       .catch(next);
   });
 
-  app.get('/', (req, res, next) => {
+  app.use((req, res, next) => {
```
Comment (lennym) [inline pages/pil/read/index.js]: Why has this changed?

File: pages/common/content/success-messages.js (hunk 1 above)
Comment (joefitter) [reply pages/common/content/success-messages.js]: good spot - copy paste fail

File: pages/pil/review/content/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  title: 'Confirm this personal licence is up to date',
+  content: 'You confirm this licence is still required and all the animal types and procedures are up to date.',
+  buttons: {
+    submit: 'Submit confirmation',
+    canReviewSubmit: 'Endorse licence'
+  },
+  fields: {
+    comments: {
+      label: 'Comments',
+      hint: 'Comments can be seen by establishment users, as well as inspectors and Home Office team members. They will be added to the ‘Latest activity log’ of this task.'
```
Comment (lennym) [inline pages/pil/review/content/index.js]: This hint text doesn't seem right for the context.

File: pages/pil/read/index.js (hunk 2 above)
Comment (joefitter) [reply pages/pil/read/index.js]: good question 🤔

File: pages/task/read/content/pil.js [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
     },
     'resubmitted': {
+      action: {
+        review: 'Resubmit review'
+      },
       hint: {
         application: 'Your application will be sent to an NTCO for endorsement',
-        amendment: 'Your amendment will be sent to an NTCO for endorsement'
+        amendment: 'Your amendment will be sent to an NTCO for endorsement',
+        review: 'Your request will be reviewed by a named training and competency officer'
```
Comment (lennym) [inline pages/task/read/content/pil.js]: This seems weirdly inconsistent.

File: pages/task/read/content/pil.js [hunk 5]
```diff
@@ -30,9 +46,7 @@ module.exports = {
   'sticky-nav': {
     applicant: {
       application: 'Applicant',
-      amendment: 'Licence holder',
-      revocation: 'Licence holder',
-      transfer: 'Licence holder'
+      other: 'Licence holder'
```
Comment (lennym) [inline pages/task/read/content/pil.js]: 🎉

File: pages/common/content/index.js [hunk 6]
```diff
@@ -219,6 +219,10 @@ module.exports = {
     transfer: {
       title: 'Transfer in progress',
       description: 'A request to transfer this licence to another establishment has been submitted and is being reviewed. You can view or recall the licence transfer.'
+    },
+    review: {
+      title: 'Review in progress',
+      description: 'A request to review this licence has been submitted and is being reviewed. You can view or recall the review request.'
```
Comment (lennym) [inline pages/common/content/index.js]: Yo dawg, I heard you like reviews

File: pages/task/read/content/pil.js (hunk 4 above)
Comment (joefitter) [reply pages/task/read/content/pil.js]: don't shoot the messenger

File: pages/task/read/content/pil.js (hunk 5 above)
Comment (joefitter) [reply pages/task/read/content/pil.js]: huzzah

File: pages/common/content/index.js (hunk 6 above)
Comment (joefitter) [reply pages/common/content/index.js]: 🤷‍♂

File: pages/pil/read/index.js [hunk 7]
```diff
@@ -18,6 +19,21 @@ module.exports = settings => {
...
+    if (!process.env['ENABLE_PIL_REVIEW']) {
+      return next();
+    }
+    const needsReview = moment(req.pil.reviewDate).isBefore(moment().add(3, 'months'));
+    if (needsReview) {
+      res.locals.static.pilReviewRequired = {
+        reviewUrl: req.buildRoute('pil.review'),
+        overdue: moment(req.pil.reviewDate).isBefore(moment())
+      };
+    }
```
Comment (lennym) [inline pages/pil/read/index.js]: I think this work might be better done API side.

Eventually we're going to want to make API calls which list/count reviews that are due, and so having the logic all in one place makes more sense.

Comment (joefitter) [reply pages/pil/read/index.js]: you're not wrong

File: pages/pil/review/index.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      req.user.can('pil.update', params),
+      req.user.can('pil.review', params)
+    ])
+      .then(([canUpdate, canReview]) => {
+        if (!canUpdate && !canReview) {
+          return next(new UnauthorisedError());
+        }
+        res.locals.static.canReview = canReview;
+        if (canReview) {
+          res.locals.static.content.buttons.submit = res.locals.static.content.buttons.canReviewSubmit;
```
Comment (lennym) [inline pages/pil/review/index.js]:
```suggestion
          // change the button text if the user can complete the review without approval
          res.locals.static.content.buttons.submit = res.locals.static.content.buttons.canReviewSubmit;
```

Comment (joefitter) [reply pages/pil/review/index.js]: thanks friend

File: pages/pil/routes.js [hunk 9]
```diff
@@ -4,11 +4,12 @@ const remove = require('./delete');
 const pdf = require('./pdf');
 const read = require('./read');
 const create = require('./create');
+const review = require('./review');
 
 module.exports = {
   read: {
     path: '/:pilId',
-    permisions: 'pil.read',
+    permissions: 'pil.read',
```
Comment (lennym) [inline pages/pil/routes.js]: Whoops!

Comment (joefitter) [reply pages/pil/routes.js]: DERP

File: pages/task/read/views/models/pil.jsx [hunk 10]
```diff
@@ -14,11 +14,16 @@ import Modules from '../../../../profile/read/views/modules';
 export default function PIL({ task, values }) {
   const profile = useSelector(state => state.static.profile);
   const establishment = useSelector(state => state.static.establishment);
-  const pil = task.data.action === 'update-conditions' ? values : task.data.data;
+  let pil = task.data.action === 'update-conditions' ? values : task.data.data;
```
Comment (lennym) [inline pages/task/read/views/models/pil.jsx]: Can you move this line down so it sits alongside the `if (isReview)`? 👇

File: pages/task/read/views/models/pil.jsx [hunk 11]
```diff
@@ -14,11 +14,16 @@ import Modules from '../../../../profile/read/views/modules';
...
   const over18 = profile.dob ? differenceInYears(new Date(), new Date(profile.dob)) >= 18 : 'unknown';
-  const isTransfer = task.data.action === 'transfer';
+  const isTransfer = task.type === 'transfer';
+  const isReview = task.type === 'review';
 
-  const applicantKey = `applicant.${task.type}`;
+  if (isReview) {
```
Comment (lennym) [inline pages/task/read/views/models/pil.jsx]:
```suggestion
  let pil = task.data.action === 'update-conditions' ? values : task.data.data;
  if (isReview) {
```

File: pages/task/read/views/models/pil.jsx (hunk 10 above)
Comment (lennym) [reply pages/task/read/views/models/pil.jsx]:
```suggestion
```

File: pages/task/read/views/models/pil.jsx [hunk 12]
```diff
@@ -125,44 +130,52 @@ export default function PIL({ task, values }) {
...
-                    exemption.species && exemption.species.length > 0 &&
-                      (
-                        <Fragment>
-                          { exemption.species.map((s, index) => (
-                            <p key={index}>{s}</p>
-                          ))}
-                        </Fragment>
-                      )
+    (
+      (!isReview || profile.certificates.length > 0) && (
```
Comment (lennym) [inline pages/task/read/views/models/pil.jsx]: Can we consolidate this condition into a `const showTraining = !isReview || profile.certificates.length > 0;` and use that in the various place have this line?

File: pages/task/read/routers/read.js [hunk 13]
```diff
@@ -145,31 +145,6 @@ module.exports = () => {
...
-
-    if (model === 'establishment' && req.task.data.action === 'update') {
-      req.task.type = 'amendment';
-      return next();
-    }
-
-    req.task.type = get(req.task, 'data.modelData.status') === 'active' ? 'amendment' : 'application';
-
-    next();
-  });
```
Comment (lennym) [inline pages/task/read/routers/read.js]: This is all the stuff that got moved to a workflow decorator, right?

File: pages/task/content/status.js [hunk 14]
```diff
@@ -200,12 +209,14 @@ module.exports = {
     action: {
       application: 'Edit and resubmit the application',
       amendment: 'Edit and resubmit the amendment',
-      transfer: 'Edit and resubmit the transfer request'
+      transfer: 'Edit and resubmit the transfer request',
+      review: 'Edit and resubmit the review request'
```
Comment (lennym) [inline pages/task/content/status.js]: Is that a thing or is this content just here for completeness?

Comment (joefitter) [reply pages/task/content/status.js]: not a thing, will remove

File: pages/task/read/views/models/pil.jsx (hunk 10 above)
Comment (joefitter) [reply pages/task/read/views/models/pil.jsx]: I sure can

File: pages/task/read/routers/read.js (hunk 13 above)
Comment (joefitter) [reply pages/task/read/routers/read.js]: sure is

**Pull request merged**